### PR TITLE
Update AutoML API

### LINKED ISSUE #24839: Update How to use the ML.NET automated ML API
- Add ranking
- Consider adding FLAML / NNI information
- Review outdated content and update as necessary

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c3bb37d9-1c4d-5d5f-2ed6-2ef77f2cd627
* Version Independent ID: 80ed0d4d-63d3-49f0-ae42-d1f132137352
* Content: [How to use the ML.NET automated ML API - ML.NET](https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/how-to-use-the-automl-api)
* Content Source: [docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md](https://github.com/dotnet/docs/blob/main/docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md)
* Product: **dotnet-ml**
* GitHub Login: @luisquintanilla
* Microsoft Alias: **luquinta**

### PR DESCRIPTION
**Pull request by jwood803**
- Add ranking to the AutoML API doc

Fixes #24839

[Preview](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/how-to-use-the-automl-api?branch=pr-en-us-25075)

AB#1864079

### CODE REVIEW

**Comment by jwood803** [comment]
@luisquintanilla I believe this is ready for review.

File: docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md [hunk 1]
```diff
@@ -12,6 +12,15 @@ Automated machine learning (AutoML) automates the process of applying machine le
 > [!NOTE]
 > This topic refers to the automated machine learning API for ML.NET, which is currently in preview. Material may be subject to change.
 
+## AutoML Implementation
```
Comment (luisquintanilla) [inline docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md:15]: @briacht how does this section sound?

File: docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md [hunk 2]
```diff
@@ -12,6 +12,15 @@ Automated machine learning (AutoML) automates the process of applying machine le
...
+
+AutoML has collaborated with Microsoft Research teams [Neural Network Intelligence](https://github.com/Microsoft/nni) (NNI) and [Fast and Lightweight AutoML](https://github.com/Microsoft/flaml) (FLAML). These teams help provide additional benefits to the AutoML implementation. These benefits include:
```
Comment (luisquintanilla) [inline docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md:17]: @jwood803 looking at this section, while technically it is true that AutoML in tooling like Model Builder leverages NNI / FLAML, I don't think those implementations are part of the AutoML API. @LittleLittleCloud can you confirm?

**Review by luisquintanilla — COMMENTED** [review]
@jwood803 looks good. Added a few comments. Waiting for confirmation to see whether NNI / FLAML section should be included in this article.

File: docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md (hunk 2 above)
Comment (briacht) [inline docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md:17]: AutoML is the feature, so I would re-word this to something like:

"The ML.NET team is collaborating with Microsoft Research Teams NNI and FLAML to improve the AutoML experience in ML.NET"

Comment (briacht) [reply docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md:17]: Nope not part of AutoML API yet

File: docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md [hunk 3]
```diff
@@ -12,6 +12,15 @@ Automated machine learning (AutoML) automates the process of applying machine le
...
+
+- Enabling AutoML support for all ML.NET scenarios
+- Allowing more precise control over the hyperparameter search space
+- Enabling more training environments, including local, Azure, and on-prem distributed training
+- Opening up future collaborations on advanced ML tech, like Network Architecture Search (NAS)
```
Comment (briacht) [inline docs/machine-learning/how-to-guides/how-to-use-the-automl-api.md:22]: Maybe change from ML to Machine Learning

**Review by luisquintanilla — APPROVED** [review]
LGTM. Thanks @jwood803

**Pull request merged**